spec/numberformat.html

```diff
@@ -104,54 +104,99 @@
+              1. If the value of the _numberFormat_.[[useGrouping]] is *true*,
+                1. Let _groupSepSymbol_ be the ILND String representing the grouping separator.
+                1. Let _groups_ be a List whose elements are, in left to right order, the substrings defined by ILND set of locations within the _integer_.
+                1. Assert: The number of elements in _groups_ List is greater than *0*.
+                1. Repeat, while _groups_ List is not empty:
+                  1. Remove the first element from _groups_ and let _integerGroup_ be the value of that element.
+                  1. Add new part record { [[type]]: *"integer"*, [[value]]: _integerGroup_ } as a new element of the list _result_.
+                  1. If _groups_ List is not empty, then:
+                    1. Add new part record { [[type]]: *"group"*, [[value]]: _groupSepSymbol_ } as a new element of the list _result_.
+              1. Else,
+                1. Add new part record { [[type]]: *"integer"*, [[value]]: _integer_ } as a new element of the list _result_.
+              1. If _fraction_ is not *undefined*, then:
+                1. Let _decimalSepSymbol_ be the ILND String representing the decimal separator.
+                1. Add new part record { [[type]]: *"decimal"*, [[value]]: _decimalSepSymbol_ } as a new element of the list _result_.
+                1. Add new part record { [[type]]: *"fraction"*, [[value]]: _fraction_ } as a new element of the list _result_.
+          1. Else if _p_ is equal *"plusSign"*, then:
+            1. Let _plusSignSymbol_ be the ILND String representing the plus sign.
+            1. Add new part record { [[type]]: *"plusSign"*, [[value]]: _plusSignSymbol_ } as a new element of the list _result_.
+          1. Else if _p_ is equal *"minusSign"*, then:
+            1. Let _minusSignSymbol_ be the ILND String representing the minus sign.
+            1. Add new part record { [[type]]: *"minusSign"*, [[value]]: _minusSignSymbol_ } as a new element of the list _result_.
+          1. Else if _p_ is equal *"percentSign"* and _numberFormat_.[[style]] is *"percent"*, then:
+            1. Let _percentSignSymbol_ be the ILND String representing the percent sign.
+            1. Add new part record { [[type]]: *"percentSign"*, [[value]]: _percentSignSymbol_ } as a new element of the list _result_.
+          1. Else if _p_ is equal *"currency"* and _numberFormat_.[[style]] is *"currency"*, then:
+            1. Let _currency_ be the value of _numberFormat_.[[currency]].
+            1. If _numberFormat_.[[currencyDisplay]] is *"code"*, then
+              1. Let _cd_ be _currency_.
+            1. Else if _numberFormat_.[[currencyDisplay]] is *"symbol"*, then
+              1. Let _cd_ be an ILD string representing _currency_ in short form. If the implementation does not have such a representation of _currency_, use _currency_ itself.
+            1. Else if _numberFormat_.[[currencyDisplay]] is *"name"*, then
```

Comment:
there is a missing code-patch here... either an assertion that there is always a value for currencyDisplay, or an else with the right value.